deployments/scroll/usdc/migrations/1706149385_configurate_and_ens.ts

```diff
@@ -0,0 +1,227 @@
+    const officialMarketsJSON = JSON.parse(await ENSResolver.text(subdomainHash, ENSTextRecordKey));
+    if (officialMarketsJSON[scrollChainId]) {
+      officialMarketsJSON[scrollChainId].push(newMarketObject);
+    } else {
+      officialMarketsJSON[scrollChainId] = [newMarketObject];
+    }
+
+    const configuration = await getConfigurationStruct(deploymentManager);
+
+    const setConfigurationCalldata = await calldata(
+      configurator.populateTransaction.setConfiguration(comet.address, configuration)
+    );
+    const deployAndUpgradeToCalldata = utils.defaultAbiCoder.encode(
+      ['address', 'address'],
+      [configurator.address, comet.address]
+    );
+    const setRewardConfigCalldata = utils.defaultAbiCoder.encode(
+      ['address', 'address'],
+      [comet.address, scrollCOMPAddress]
+    );
+    const l2ProposalData = utils.defaultAbiCoder.encode(
+      ['address[]', 'uint256[]', 'string[]', 'bytes[]'],
+      [
+        [configurator.address, cometAdmin.address, rewards.address],
+        [0, 0, 0],
+        [
+          'setConfiguration(address,(address,address,address,address,address,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint64,uint104,uint104,uint104,(address,address,uint8,uint64,uint64,uint64,uint128)[]))',
+          'deployAndUpgradeTo(address,address)',
+          'setRewardConfig(address,address)'
+        ],
+        [setConfigurationCalldata, deployAndUpgradeToCalldata, setRewardConfigCalldata]
+      ]
+    );
+
+    const COMPAmountToBridge = exp(10_000, 18);
+
+    const actions = [
+      // 1. Set Comet configuration + deployAndUpgradeTo new Comet and set reward config on Scroll
+      {
+        contract: scrollMessenger,
+        signature: 'sendMessage(address,uint256,bytes,uint256)',
+        args: [bridgeReceiver.address, 0, l2ProposalData, 600_000]
+      },
+
+      // 2. Approve Ethereum's StandardERC20Gateway to take Timelock's COMP (for bridging)
+      {
+        contract: COMP,
+        signature: 'approve(address,uint256)',
+        args: [scrollL1StandardERC20GatewayAddress, COMPAmountToBridge]
+      },
+      // 3. Bridge COMP from Ethereum to Scroll Comet using L1GatewayRouter
+      {
+        contract: scrollL1TokenBridge,
+        signature: 'depositERC20(address,address,uint256,uint256)',
+        args: [COMP.address, rewards.address, COMPAmountToBridge, 300_000]
+      },
+      // 4. Update the list of official markets
+      {
+        target: ENSResolverAddress,
+        signature: 'setText(bytes32,string,string)',
+        calldata: ethers.utils.defaultAbiCoder.encode(
+          ['bytes32', 'string', 'string'],
+          [subdomainHash, ENSTextRecordKey, JSON.stringify(officialMarketsJSON)]
+        )
+      }
+    ];
+
+    const description =
+      '# Configurate Scroll cUSDCv3 market, set reward config, bridge over USDC and COMP, and update ENS text record.';
```

Comment:
Probably would want to flesh this out a bit more, with more context and link to discussions on forums.

Reply:
updated with more detail!